### PR TITLE
Support async bag download

### PR DESCRIPTION
**Pull request by pkdash**

### CODE REVIEW

**Comment by pkdash** [comment]
@hyi can you review this small PR? After this gets merged then only I can publish to pypi.

File: hs_restclient/__init__.py [hunk 1]
```diff
@@ -477,16 +482,19 @@ def getResource(self, pid, destination=None, unzip=False):
...
             directory named $(PID) residing in destination. Only applies when destination is not None.
 
+        :param wait_for_bag_creation: True if to wait to download the bag in case the bag is not ready
+            (bag needs to be recreated before it can be downloaded).
         :raises: HydroShareArgumentException if any arguments are invalid.
         :raises: HydroShareNotAuthorized if the user is not authorized to access the
             resource.
         :raises: HydroShareNotFound if the resource was not found.
         :raises: HydroShareHTTPException to signal an HTTP error
+        :raise: HydroShareBagNotReady if the bag is not reday to be downloaded and wait_for_bag_download is False
```
Comment (hyi) [inline hs_restclient/__init__.py]: @pkdash correct the typo "reday" to "ready". Also I am not sure about the ```wait_for_bag_download is False``` part above. if ```wait_for_bag_download is False``` I am thinking ```HydroShareBagNotReady``` exception should not be raised, right?

Comment (hyi) [inline hs_restclient/__init__.py]: @pkdash Also, ```wait_for_bag_download``` should be ```wait_for_bag_creation```, right?

Comment (pkdash) [inline hs_restclient/__init__.py]: @hyi good catch on parameter name wrong in the documentation. The exception should be raised if the user says not to wait and we find the bag is not ready.

Comment (hyi) [inline hs_restclient/__init__.py]: @pkdash I got the part why the exception should be raised now. Just correct the two typos as I noted above.

**Comment by hyi** [comment]
@pkdash Only a minor comment for you to correct typos in docstring. After that is done, a +1 from me.

**Comment by pkdash** [comment]
@hyi can I then merge?

**Comment by hyi** [comment]
@pkdash Yes, you can merge.

**Pull request merged**